ethereum/contracts/SwapCreator.sol

```diff
@@ -190,45 +205,55 @@ contract SwapCreator is ERC2771Context, Secp256k1 {
 
     // Bob can claim if:
     // - (Alice has set the swap to `ready` or it's past timeout0) and it's before timeout1
-    // This function is only callable by the trusted forwarder.
-    // It transfers the fee to the originator of the transaction.
-    function claimRelayer(Swap memory _swap, bytes32 _s, uint256 fee) public {
-        if (!isTrustedForwarder(msg.sender)) revert OnlyTrustedForwarder();
-        _claim(_swap, _s);
+    // It transfers the fee to the relayer address specified in `_relaySwap`.
+    // Note: this function will revert if the swa value is less than the relayer fee;
+    // in that case, `claim` must be called instead.
+    function claimRelayer(
+        RelaySwap memory _relaySwap,
+        bytes32 _secret,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public {
+        address signer = ecrecover(keccak256(abi.encode(_relaySwap)), v, r, s);
+        if (signer != _relaySwap.swap.claimer) revert InvalidSignature();
+        if (address(this) != _relaySwap.swapCreator) revert InvalidContractAddress();
+
+        _claim(_relaySwap.swap, _secret);
 
         // send ether to swap claimer, subtracting the relayer fee
-        // which is sent to the originator of the transaction.
-        // tx.origin is okay here, since it isn't for authentication purposes.
-        if (_swap.asset == address(0)) {
-            _swap.claimer.transfer(_swap.value - fee);
-            payable(tx.origin).transfer(fee); // solhint-disable-line
+        if (_relaySwap.swap.asset == address(0)) {
+            _relaySwap.swap.claimer.transfer(_relaySwap.swap.value - _relaySwap.fee);
+            payable(_relaySwap.relayer).transfer(_relaySwap.fee); // solhint-disable-line
         } else {
             // WARN: this will FAIL for fee-on-transfer or rebasing tokens if the token
             // transfer reverts (i.e. if this contract does not contain _swap.value tokens),
             // exposing Bob's secret while giving him nothing.
-            IERC20(_swap.asset).transfer(_swap.claimer, _swap.value - fee);
-            IERC20(_swap.asset).transfer(tx.origin, fee); // solhint-disable-line
+            IERC20(_relaySwap.swap.asset).transfer(
+                _relaySwap.swap.claimer,
+                _relaySwap.swap.value - _relaySwap.fee
+            );
+            IERC20(_relaySwap.swap.asset).transfer(_relaySwap.relayer, _relaySwap.fee); // solhint-disable-line
```

Comment:
Do we want this `else` branch in the contract now, when our code does not currently allow relaying ERC20 swaps? (Just asking, I'm not against it, but it might confuse someone into thinking that we support it.)

Reply:
idk, doesn't hurt to have it, better than redeploying the contract if we decide to support it

Reply:
I'm wondering if we might end up deploying several times before figuring out how to help a user that wants ERC20 tokens in an account that has no ETH to do anything with the tokens.  :)

Reply:
with account abstraction, they should technically be able to use ERC20s in an account with no tokens, but yeah the user needs to be aware of that lol